src/controllers/controller.doughnut.js

```diff
@@ -45,20 +45,14 @@ defaults._set('doughnut', {
 				if (data.labels.length && data.datasets.length) {
 					return data.labels.map(function(label, i) {
 						var meta = chart.getDatasetMeta(0);
-						var ds = data.datasets[0];
-						var arc = meta.data[i];
-						var custom = arc && arc.custom || {};
-						var arcOpts = chart.options.elements.arc;
-						var fill = resolve([custom.backgroundColor, ds.backgroundColor, arcOpts.backgroundColor], undefined, i);
-						var stroke = resolve([custom.borderColor, ds.borderColor, arcOpts.borderColor], undefined, i);
-						var bw = resolve([custom.borderWidth, ds.borderWidth, arcOpts.borderWidth], undefined, i);
+						var model = meta.controller._resolveElementOptions(meta.data[i] || {}, i);
```

Comment:
Should we use `var model = meta.data[i]._options` instead? (as [defined here](/chartjs/Chart.js/blob/master/src/controllers/controller.doughnut.js#L177))
Or maybe `var model = meta.data[i]._model`?

Reply:
There is a change that a dataset has no data but styles. `_model` cannot be used if there is no data in the dataset.

Reply:
You right